lib/src/models/subreddit_moderation.dart

```diff
@@ -0,0 +1,647 @@
+    case 'editrule':
+      return ModeratorActionType.editRule;
+    case 'editsettings':
+      return ModeratorActionType.editSettings;
+    case 'ignorereports':
+      return ModeratorActionType.ignoreReports;
+    case 'invitemoderator':
+      return ModeratorActionType.inviteModerator;
+    case 'lock':
+      return ModeratorActionType.lock;
+    case 'marknsfw':
+      return ModeratorActionType.markNSFW;
+    case 'modmail_enrollment':
+      return ModeratorActionType.modmailEnrollment;
+    case 'muteuser':
+      return ModeratorActionType.muteUser;
+    case 'removecomment':
+      return ModeratorActionType.removeComment;
+    case 'removecontributor':
+      return ModeratorActionType.removeContributor;
+    case 'removelink':
+      return ModeratorActionType.removeLink;
+    case 'removemoderator':
+      return ModeratorActionType.removeModerator;
+    case 'removewikicontributor':
+      return ModeratorActionType.removeWikiContributor;
+    case 'setcontestmode':
+      return ModeratorActionType.setContestMode;
+    case 'setpermissions':
+      return ModeratorActionType.setPermissions;
+    case 'setsuggestedsort':
+      return ModeratorActionType.setSuggestedSort;
+    case 'spamcomment':
+      return ModeratorActionType.spamComment;
+    case 'spamlink':
+      return ModeratorActionType.spamLink;
+    case 'spoiler':
+      return ModeratorActionType.spoiler;
+    case 'sticky':
+      return ModeratorActionType.sticky;
+    case 'unbanuser':
+      return ModeratorActionType.unbanUser;
+    case 'unignorereports':
+      return ModeratorActionType.unignoreReports;
+    case 'uninvitemoderator':
+      return ModeratorActionType.uninviteModerator;
+    case 'unlock':
+      return ModeratorActionType.unlock;
+    case 'unmuteuser':
+      return ModeratorActionType.unmuteUser;
+    case 'unsetcontestmode':
+      return ModeratorActionType.unsetContestMode;
+    case 'unspoiler':
+      return ModeratorActionType.unspoiler;
+    case 'unsticky':
+      return ModeratorActionType.unsticky;
+    case 'wikibanned':
+      return ModeratorActionType.wikiBanned;
+    case 'wikicontributor':
+      return ModeratorActionType.wikiContributor;
+    case 'wikipagelisted':
+      return ModeratorActionType.wikiPageListed;
+    case 'wikipermlevel':
+      return ModeratorActionType.wikiPermLevel;
+    case 'wikirevise':
+      return ModeratorActionType.wikiRevise;
+    case 'wikiunbanned':
+      return ModeratorActionType.wikiUnbanned;
+    default:
+      throw new DRAWInternalError("'$s' is not a valid moderator action type.");
+  }
+}
+
+class SubredditSettings {
+  final Map _data;
+  final SubredditRef subreddit;
+
+  SubredditSettings.copy(SubredditSettings original)
+      : _data = new Map.from(original._data),
+        subreddit = original.subreddit;
+
+  SubredditSettings._(this.subreddit, this._data);
+
+  bool get defaultSet => _data['default_set'];
+  void set defaultSet(bool x) => _data['default_set'] = x;
```

Comment:
Here's an [example](https://api.dartlang.org/stable/1.24.3/dart-io/exitCode.html) of what dartdoc generates for getters/setters. In this case, I don't think changing `x` to another name adds much value since it's a very simple setter and it's clear what is happening.